src/Credentials/AssumeRoleWithWebIdentityCredentialProvider.php

```diff
@@ -114,7 +114,7 @@ public function __invoke()
                     }
                     if (empty($token)) {
                         if ($this->tokenFileReadAttempts < $this->retries) {
-                            sleep(pow(1.2, $this->tokenFileReadAttempts));
+                            sleep((int) pow(1.2, $this->tokenFileReadAttempts));
```

Comment:
I've kept the result of the code the same, but it does lose some precision which might be wanted.

An alternative fix would be using `usleep()`, to keep a higher resolution.

```php 
usleep((int) (1000*1000*pow(1.2, $this->tokenFileReadAttempts)));
```
(Same goes for the same change in other places where this was changed.)

Reply:
I do not know if you have seen but in the file aws-sdk-php/src/Credentials/InstanceProfileProvider.php there is the same fix to do in line 262 : 

```
    private function handleRetryableException(
        \Exception $e,
        $retryOptions,
        $message
    ) {
        $isRetryable = true;
        if (!empty($status = $this->getExceptionStatusCode($e))
            && isset($retryOptions['blacklist'])
            && in_array($status, $retryOptions['blacklist'])
        ) {
            $isRetryable = false;
        }
        if ($isRetryable && $this->attempts < $this->retries) {
            sleep((int) pow(1.2, $this->attempts));            <-----
        } else {
            throw new CredentialsException($message);
        }
    }
```